libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/OAuth/CSFSalesforceOAuthRefresh.m

```diff
@@ -97,6 +97,12 @@ - (void)oauthCoordinatorDidAuthenticate:(SFOAuthCoordinator *)coordinator authIn
     self.network.account.credentials = coordinator.credentials;
     CSFOAuthTokenRefreshOutput *output = [[CSFOAuthTokenRefreshOutput alloc] initWithCoordinator:coordinator];
     [self finishWithOutput:output error:nil];
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter] postNotificationName:kSFAuthenticationManagerFinishedNotification
```

Comment:
This and one other change were a collaboration between @jhneves and myself. I don't remember the reasoning at the time for posting to the main thread and I agree it shouldn't need to be explicitly posted to the main thread.